app/src/main/java/ingsw/group1/findmyphone/ResponseManager.java

```diff
@@ -0,0 +1,182 @@
+package ingsw.group1.findmyphone;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.MutableLiveData;
+
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+
+import ingsw.group1.findmyphone.activity.NavHolderActivity;
+import ingsw.group1.findmyphone.contacts.SMSContactManager;
+import ingsw.group1.findmyphone.event.EventType;
+import ingsw.group1.findmyphone.event.SMSLogDatabase;
+import ingsw.group1.findmyphone.event.SMSLogEvent;
+import ingsw.group1.findmyphone.location.GeoPosition;
+import ingsw.group1.findmyphone.location.LocationManager;
+import ingsw.group1.findmyphone.location.LocationResponseCommand;
+import ingsw.group1.findmyphone.log.LogManager;
+import ingsw.group1.findmyphone.log.items.LogItem;
+import ingsw.group1.findmyphone.log.items.LogItemFormatter;
+
+/**
+ * Class meant to manage responses to the various incoming messages. This includes starting the
+ * appropriate activities/services if necessary.
+ *
+ * @author Riccardo De Zen.
+ */
+public class ResponseManager {
+
+    private static ResponseManager instance;
+
+    private LocationManager locationManager = new LocationManager();
+    private MessageParser messageParser = new MessageParser();
+    private SMSContactManager contacts;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'contacts' could be replaced by a local variable.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5114829)